### ISSUE TITLE
Fix S2583 FP: false 'always true' condition inside of loop when loop limit is a constant

### ISSUE THREAD

**Issue by eduherminio** [SonarSource/sonar-dotnet#8248]
### Description

Please provide a succinct description of your issue. Also please provide the rule ID (e.g. S1234)

### Repro steps

Given:
```csharp
public static class Constants
{
    public const int AbsoluteMaxDepth = 255;
}
```

This code is a FP

```csharp
        for (int searchDepth = 1; searchDepth < Constants.AbsoluteMaxDepth; ++searchDepth) 
        {
            for (int movesSearchedCount = 1; movesSearchedCount < 255; ++movesSearchedCount)
            {
                var depthFactor = searchDepth < 10 ? 1 : 2 // S2583 
            }
        }
```

This code is fine:

```csharp
        for (int searchDepth = 1; searchDepth < 255; ++searchDepth) 
        {
            for (int movesSearchedCount = 1; movesSearchedCount < 255; ++movesSearchedCount)
            {
                var depthFactor = searchDepth < 10 ? 1 : 2 // S2583 
            }
        }
```

### Expected behavior

No FP

### Actual behavior

FP

### Related information
- .NET: 8.0.100-rc.2.23502.2
- VS Sonar extension: 7.3.0.77872
- VS: Microsoft Visual Studio Enterprise 2022 (64-bit) - Version 17.7.0
- Windows 10

**Comment by Tim-Pohlmann** [comment]
Hello @eduherminio!
Thank you very much for your report. I have good news! This issue will be fixed in the next release.
I close this issue as a duplicate of #8080. Please let us know if you still encounter this issue when the new version is released and you have upgraded.

**Comment by eduherminio** [comment]
Cheers @Tim-Pohlmann, I somehow failed to find the duplicate, should have looked also into the closed ones!

**Comment by Tim-Pohlmann** [comment]
Don't worry about it. It's very easy to miss an existing issue that is closed but has yet to be released. It's just unfortunate timing.
